source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -1027,48 +1027,16 @@ void ShaderGraph::optimize(GenContext& context)
         }
         else if (node->hasClassification(ShaderNode::Classification::DOT))
         {
-            // Dot nodes without modifiers can be elided by moving their connection downstream.
+            // Filename dot nodes must be elided so they do not create extra samplers.
             ShaderInput* in = node->getInput("in");
-            if (in->getChannels().empty())
+            if (in->getChannels().empty() && in->getType() == Type::FILENAME)
             {
                 bypass(context, node, 0);
                 ++numEdits;
             }
         }
-        else if (node->hasClassification(ShaderNode::Classification::IFELSE))
-        {
-            // Check if we have a constant conditional expression
-            ShaderInput* intest = node->getInput("intest");
-            if (!intest->getConnection())
-            {
-                // Find which branch should be taken
-                ShaderInput* cutoff = node->getInput("cutoff");
-                ValuePtr value = intest->getValue();
-                const float intestValue = value ? value->asA<float>() : 0.0f;
-                const int branch = (intestValue <= cutoff->getValue()->asA<float>() ? 2 : 3);
-
-                // Bypass the conditional using the taken branch
-                bypass(context, node, branch);
-
-                ++numEdits;
-            }
-        }
-        else if (node->hasClassification(ShaderNode::Classification::SWITCH))
```

Comment:
This worked in 1.37 where the "which" input was a "parameter" instead of an "input". Since this input is not marked as "uniform", it should not be elided.

Reply:
I think this should be restored as it checks to see if indeed a uniform unconnected value. I have log an new issue and can discuss there, but now new shaders will always add more code in.  (https://github.com/AcademySoftwareFoundation/MaterialX/issues/1526)